src/main/java/org/broadinstitute/dsm/route/KitUploadRoute.java

```diff
@@ -343,74 +350,102 @@ private void addKitRequest(Connection conn, String kitTypeName, KitRequestSettin
+                throw new RuntimeException("user with shortId: " + shortId + " and name: " +
+                        participantFirstNameFromDoc + " " + participantLastNameFromDoc + " does not belong to this study.");
+            }
+
+            KitUploadObject participantKitToUpload;
+            if (nameInOneColumn) {
+                participantKitToUpload = new KitUploadObject(null, participantDataByFieldName.get(PARTICIPANT_ID), shortId,
+                        null, participantDataByFieldName.get(SIGNATURE),
+                        participantDataByFieldName.get(STREET1), participantDataByFieldName.get(STREET2), participantDataByFieldName.get(CITY),
+                        participantDataByFieldName.get(STATE), participantDataByFieldName.get(POSTAL_CODE), participantDataByFieldName.get(COUNTRY), participantDataByFieldName.getOrDefault(PHONE_NUMBER, null));
             }
+            else if (containsOrderNumber) {
+                participantKitToUpload = new KitUploadObject(participantDataByFieldName.get(ORDER_NUMBER), participantDataByFieldName.get(PARTICIPANT_ID), null,
+                        null, participantDataByFieldName.get(NAME),
+                        participantDataByFieldName.get(STREET1), participantDataByFieldName.get(STREET2), participantDataByFieldName.get(CITY),
+                        participantDataByFieldName.get(STATE), participantDataByFieldName.get(POSTAL_CODE), participantDataByFieldName.get(COUNTRY), participantDataByFieldName.getOrDefault(PHONE_NUMBER, null));
+            }
+            else {
+                participantKitToUpload = new KitUploadObject(null, participantDataByFieldName.get(PARTICIPANT_ID), shortId,
+                        participantDataByFieldName.get(FIRST_NAME), participantDataByFieldName.get(LAST_NAME),
+                        participantDataByFieldName.get(STREET1), participantDataByFieldName.get(STREET2), participantDataByFieldName.get(CITY),
+                        participantDataByFieldName.get(STATE), participantDataByFieldName.get(POSTAL_CODE), participantDataByFieldName.get(COUNTRY), participantDataByFieldName.getOrDefault(PHONE_NUMBER, null));
+            }
+            kitRequestsToUpload.add(participantKitToUpload);
+
         }
-        return null;
+    }
+
+    private boolean userExistsInRealm(String shortId, String realm, String participantFirstNameFromDoc,
+                                      String participantLastNameFromDoc) {
+        if (StringUtils.isBlank(shortId) && shortId.matches("^[a-zA-Z0-9]*$")) {
+            return false;
+        }
+        if (participantFirstNameFromDoc == null && participantLastNameFromDoc == null) return false;
+
+        Map<String, String> queryConditions = new HashMap<>();
+        queryConditions.put("ES", " AND profile.hruid = '" + shortId + "'");
+        DDPInstance ddpInstanceByRealm = DDPInstance.getDDPInstance(realm);
```

Comment:
can be moved further out and only the DDPInstance given into the method, because DDPInstance will always be the same for all pts in the upload so no need to get the instance for every participant